src/vm/interoputil.cpp

```diff
@@ -5095,11 +5095,11 @@ ClassFactoryBase *GetComClassFactory(MethodTable* pClassMT)
     if (pClsFac == NULL)
     {
         //
-        // Collectible types do not support com interop
+        // Collectible types do not support WinRT interop
         //
-        if (pClassMT->Collectible())
+        if (pClassMT->IsExportedToWinRT() || pClassMT->IsProjectedFromWinRT())
```

Comment:
I don't understand this change. Directly below on lines 5107 and 5114 we check for both of these and handle them without throwing. What am I missing?

Reply:
Sigh, it is missing check for Collectible - it should have been an addition to the condition and not a replacement.